lxd/instance/drivers/driver_qemu.go

```diff
@@ -7427,21 +7442,65 @@ func (d *qemu) DeviceEventHandler(runConf *deviceConfig.RunConfig) error {
 }
 
 // vsockID returns the vsock Context ID for the VM.
-func (d *qemu) vsockID() int {
-	// We use the system's own VsockID as the base.
-	//
-	// This is either "2" for a physical system or the VM's own id if
-	// running inside of a VM.
-	//
-	// To this we add 1 for backward compatibility with prior logic
-	// which would start at id 3 rather than id 2. Removing that offset
-	// would cause conflicts between existing VMs until they're all rebooted.
-	//
-	// We then add the VM's own instance id (1 or higher) to give us a
-	// unique, non-clashing context ID for our guest.
+func (d *qemu) vsockID() (int, error) {
+	// Return an already existing Context ID.
+	existingVsockID, ok := d.localConfig["volatile.vsock_id"]
+	if ok {
+		vsockID, err := strconv.Atoi(existingVsockID)
+		if err != nil {
+			return 0, fmt.Errorf("Failed to convert Context ID: %w", err)
+		}
 
-	info := DriverStatuses()[instancetype.VM].Info
-	return info.Features["vhost_vsock"].(int) + 1 + d.id
+		return vsockID, nil
+	}
+
+	instanceUUID := uuid.Parse(d.localConfig["volatile.uuid"])
+	if instanceUUID == nil {
+		return 0, fmt.Errorf("Instance does not have a UUID")
+	}
+
+	// Pick the first 4 bytes of the random UUID (v4) to create an uint32 representation.
+	// Since the instances UUID already exist, we don't need to create another random value.
+	firstBytes := binary.LittleEndian.Uint32(instanceUUID[:4])
+
+	var overflow bool
+
+	// Try to find a new Context ID.
+	// Since the first guess might already be a collision (only 4 bytes out of 16), try again.
+	// If the candidate vsock ID overflows, it just starts all over but skipping 0-2.
+	for i := uint32(0); (firstBytes+i >= firstBytes && !overflow) || firstBytes+i < firstBytes; i++ {
+		candidateVsockID := firstBytes + i
+
+		if candidateVsockID == 0 {
+			overflow = true
+		}
+
+		// Don't try to acquire the reserved Context IDs 0-2
+		if candidateVsockID <= 2 {
+			continue
+		}
+
+		c, err := lxdvsock.Dial(candidateVsockID, shared.HTTPSDefaultPort)
+		if err != nil {
+			var unixErrno unix.Errno
```

Comment:
The examples I've seen online always use a pointer, e.g. `var unixErrno *unix.Errno`. Have you tested this with errors other than `unix.Errno`?

Reply:
I just tested this, and with `unix.Errno` there's indeed no need for a pointer.